### PR TITLE
Rewrite `README` & Add Pangolin Local Testnet

### PR DESCRIPTION
**Pull request by AsceticBear**

### CODE REVIEW

**Comment by AsceticBear** [comment]
```sh
darwinia-common [bear-readme-pangolin] » ./tests/run-pangolin/start.sh                                                 ~/w/d/darwinia-common
===> Build node
   Compiling drml v2.5.0 (/home/bear/working/darwinia/darwinia-common/bin/node/cli)
    Finished release [optimized] target(s) in 1m 58s
===> Clean old chain data
===> Setup all the validators
start n1 ...
start n2 ...
start n3 ...
start n4 ...
start n5 ...
start n6 ...
===> The pangolin network is running successfully
```

@AurevoirXavier Something needs your help. Would you like to help me test the script on Mac OS? It works in Linux, I want to ensure it works well on the Mac.

**Review by AurevoirXavier — CHANGES_REQUESTED** [review]
You post all the secret keys online.

I'll add a local testnet for this.

**Review by AurevoirXavier — APPROVED** [review]
Are you OK with these changes? @AsceticBear

**Comment by AsceticBear** [comment]
Waiting minutes, I am reviewing.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,57 @@
-# darwinia-common
-[Substrate](https://github.com/paritytech/substrate)-like common libraries, but only for Darwinia blockchains!
+# Darwinia Runtime Modules Library
+The Darwinia Runtime Modules Library (DRML) is a darwinia.network maintained collection of Substrate runtime modules.
+
+## Runtime Modules Overview
+- [darwinia-balances](./frame/balances)
+	- Desc.
```
Comment (AsceticBear) [inline README.md:6]: I think we do not need this `Desc`. It is better to add a readme document under pallet directory if we want to add pallet description.

Comment (AurevoirXavier) [reply README.md:6]: Follow this https://github.com/open-web3-stack/open-runtime-module-library

But TODO

File: README.md [hunk 2]
```diff
@@ -1,2 +1,57 @@
...
+- [darwinia-treasury](./frame/treasury)
+	- Desc.
+- [darwinia-vesting](./frame/vesting)
+	- Desc.
+
+## Development
+
+### Deploy A Pangolin Local Testnet
+```sh
+tests/run-pangolin/start.sh
```
Comment (AsceticBear) [inline README.md:56]: update this command please

**Review by AurevoirXavier — CHANGES_REQUESTED** [review]
@robinwei-itering Please help with the `Desc` part.

File: README.md (hunk 1 above)
Comment (AsceticBear) [reply README.md:6]: See, I recommend adding a one-line description in this pull request.

**Comment by AsceticBear** [comment]
The log is empty!

image
image

File: tests/pangolin-local-testnet/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+echo "Build \`drml\`"
+cargo build --release
+
+index=0
+for validator in alice bob charlie dave eve ferdie
+do
+  echo "Purge $validator's \`db\`, \`network\`, \`dvm\`"
+  rm -rf $DATA_DIR/$validator/chains/pangolin/db
+  rm -rf $DATA_DIR/$validator/chains/pangolin/network
+  rm -rf $DATA_DIR/$validator/chains/pangolin/dvm
```
Comment (AsceticBear) [inline tests/pangolin-local-testnet/deploy.sh:25]: The node data path should be `pangolin-$validator`

**Review by AsceticBear — COMMENTED** [review]
Great work. LGTM. I am unable to add green approval since I raised the PR.

**Review by HackFisher — APPROVED** [review]

**Pull request merged**